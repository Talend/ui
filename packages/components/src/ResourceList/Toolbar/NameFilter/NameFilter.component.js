import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import DebounceInput from 'react-debounce-input';
import { useTranslation } from 'react-i18next';
import { useKey } from 'react-use';

import { Action } from '../../../Actions';
import I18N_DOMAIN_COMPONENTS from '../../../constants';

import theme from './NameFilter.scss';

function NameFilter({ label, value, onChange }) {
	const { t } = useTranslation(I18N_DOMAIN_COMPONENTS);
	const removeFilter = () => onChange({ target: { value: '' } });
	useKey('Escape', removeFilter);
	return (
		<form
			className={classNames(
				'tc-resource-picker-name-filter',
				theme['tc-resource-picker-name-filter'],
			)}
		>
			<label htmlFor="resource-picker-toolbar-name-filter-input" className="sr-only">
				{label}
			</label>
			<DebounceInput
				id="resource-picker-toolbar-name-filter-input"
				type="text"
				placeholder={label}
				value={value}
				onChange={onChange}
				className="form-control"
				autoComplete="off"
				required
			/>
			<Action
				className={theme.remove}
				bsStyle="link"
				icon="talend-cross"
				label={t('LIST_FILTER_REMOVE', { defaultValue: 'Remove filter' })}
				hideLabel
				onClick={removeFilter}
			/>
		</form>
	);
}

NameFilter.propTypes = {
	label: PropTypes.string.isRequired,
	value: PropTypes.string,
	onChange: PropTypes.func,
};

export default NameFilter;
