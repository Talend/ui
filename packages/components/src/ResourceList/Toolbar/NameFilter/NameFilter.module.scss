@use '~@talend/bootstrap-theme/src/theme/guidelines' as *;
@use '~@talend/design-tokens/lib/tokens';

.tc-resource-picker-name-filter {
	flex: 1;
	position: relative;

	input + .remove {
		position: absolute;
		top: 0;
		right: 0;
	}

	input:not(:invalid) + .remove {
		display: inline-block;
	}
}

.remove {
	display: none;
	color: tokens.$coral-color-neutral-text;
	padding: $padding-smaller 0;
	line-height: 1.25rem;
	height: 100%;

	:global(.tc-svg-icon) {
		height: $svg-xs-size;
		width: $svg-xs-size;
		margin: 0;
		vertical-align: baseline;
	}
}
