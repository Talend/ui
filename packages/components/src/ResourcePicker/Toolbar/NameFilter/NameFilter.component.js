import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import DebounceInput from 'react-debounce-input';

import theme from './NameFilter.scss';

function NameFilter({ label, value, onChange }) {
	return (
		<form
			className={classNames(
				'tc-resource-picker-name-filter',
				theme['tc-resource-picker-name-filter'],
			)}
		>
			<label htmlFor="resource-picker-toolbar-name-filter-input" className="sr-only">
				{label}
			</label>
			<DebounceInput
				id="resource-picker-toolbar-name-filter-input"
				type="text"
				placeholder={label}
				value={value}
				onChange={onChange}
				className="form-control"
				autoComplete="off"
			/>
		</form>
	);
}

NameFilter.propTypes = {
	label: PropTypes.string.isRequired,
	value: PropTypes.string,
	onChange: PropTypes.func,
};

export default NameFilter;
