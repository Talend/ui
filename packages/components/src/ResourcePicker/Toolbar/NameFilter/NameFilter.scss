.tc-resource-picker-name-filter {
	flex: 1;
	position: relative;

	input + .remove {
		position: absolute;
		top: 0;
		right: 0;
	}

	input:not(:invalid) + .remove {
		display: inline-block;
	}
}

.remove {
	display: none;
	color: opacify($gray, 0.2);
	padding: $padding-smaller 0;
	line-height: 2rem;
	height: 100%;

	:global(.tc-svg-icon) {
		height: $svg-xs-size;
		width: $svg-xs-size;
		margin: 0;
		vertical-align: baseline;
	}
}
