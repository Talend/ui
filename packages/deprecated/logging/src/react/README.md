# React error logger

This module register a React updates hook that sends a log to an endpoint.

You need to configure 2 info
* the log server url
* the app state to send

## Configuration

You need to configure the logger in order to fit your backend logging service and add your app state.

```javascript
import { store as cmfstore } from '@talend/react-cmf';
import initReactLogger from '@talend/log/lib/react';

const store = cmfstore.initialize(appReducer);
initReactLogger({
    serverUrl: 'http://localhost:8888/error',
    getState: () => store.getState(),
    processState: state => removeSensitiveData(state),
});

```

You need to remove the user data from app state.
