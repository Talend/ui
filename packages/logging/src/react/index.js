import initReactLogger from './react-logger';

export default initReactLogger;
