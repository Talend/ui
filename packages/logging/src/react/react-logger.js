import ReactUpdates from 'react-dom/lib/ReactUpdates';
import ReactDefaultBatchingStrategy from 'react-dom/lib/ReactDefaultBatchingStrategy';
import { initErrorTransformer, TraceKit } from '../api/errorTransformer';
import getStatePayloadMiddleware from '../api/payloadMiddleware';

function injectLogger() {
	let isHandlingError = false;
	const ReactTryCatchBatchingStrategy = {
		// this is part of the BatchingStrategy API. simply pass along
		// what the default batching strategy would do.
		get isBatchingUpdates() {
			return ReactDefaultBatchingStrategy.isBatchingUpdates;
		},

		batchedUpdates(...args) {
			try {
				ReactDefaultBatchingStrategy.batchedUpdates(...args);
			} catch (e) {
				if (isHandlingError) {
					// our error handling code threw an error. just throw now
					throw e;
				}

				isHandlingError = true;
				try {
					TraceKit.report(e);
				} finally {
					isHandlingError = false;
				}
			}
		},
	};

	ReactUpdates.injection.injectBatchingStrategy(ReactTryCatchBatchingStrategy);
}

export default function initReactLogger({ serverUrl, getState, processState = state => state }) {
	initErrorTransformer(serverUrl, {
		payloadMiddleware: getStatePayloadMiddleware(() => processState(getState())),
	});
	injectLogger();
}
