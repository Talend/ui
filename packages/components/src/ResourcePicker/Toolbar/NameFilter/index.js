import NameFilter from './NameFilter.component';

export default NameFilter;
