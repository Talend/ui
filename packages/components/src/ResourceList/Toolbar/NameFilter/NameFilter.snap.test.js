import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import NameFilter from './NameFilter.component';

describe('NameFilter component snaps', () => {
	it('should render', () => {
		const props = {
			onChange: () => {},
			label: 'Example label',
		};

		const { container } = render(<NameFilter {...props} />);

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should reset the filter', async () => {
		const user = userEvent.setup();

		const onChange = jest.fn();
		const props = {
			onChange,
			label: 'Example label',
		};

		render(<NameFilter {...props} />);
		await user.click(screen.getByLabelText('Remove filter'));

		expect(onChange).toHaveBeenLastCalledWith({ target: { value: '' } });
	});
});
