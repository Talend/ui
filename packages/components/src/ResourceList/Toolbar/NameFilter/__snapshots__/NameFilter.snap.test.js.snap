// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameFilter component snaps should render NameFilter in default mode 1`] = `
<form
  className="tc-resource-picker-name-filter theme-tc-resource-picker-name-filter"
>
  <label
    className="sr-only"
    htmlFor="resource-picker-toolbar-name-filter-input"
  >
    Example label
  </label>
  <DebounceInput
    autoComplete="off"
    className="form-control"
    debounceTimeout={100}
    element="input"
    forceNotifyByEnter={true}
    forceNotifyOnBlur={true}
    id="resource-picker-toolbar-name-filter-input"
    minLength={0}
    onChange={[Function]}
    placeholder="Example label"
    required={true}
    type="text"
  />
  <Action
    bsStyle="link"
    className="theme-remove"
    hideLabel={true}
    icon="talend-cross"
    label="Remove filter"
    onClick={[Function]}
  />
</form>
`;
